src/cli.rs

```diff
@@ -149,13 +196,28 @@ impl Config {
         Ok(serde_yaml::from_slice(file_contents.as_slice())?)
     }
 
-    pub fn dataset_id(&self, dataset: &str) -> Option<DatasetId> {
-        self.available_datasets
-            .get(dataset)
-            .map(DatasetId::from_url)
+    pub fn find_dataset(&self, slug: &str) -> Option<&DatasetConfig> {
+        self.available_datasets.iter().find(|d| {
```

Comment:
Even if we have about 100-200 datasets, it's probably too slow. We should use `BTreeMap` instead